bambootracker: init at 0.4.3

OPNA2608 opened the pull request:
###### Motivation for this change
Been in my local checkout for awhile, decided to push since it was mentioned in https://github.com/NixOS/nixpkgs/issues/81815.

Ping @fgaz because you were interested in packaging it. Give it a look, let me know how it looks to you.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after) (new package -> 326780408)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

OPNA2608 commented:
Seems to eval fine on x86 & ARM Linux, maybe a trusted user could kick off a Darwin build for completeness' sake? :slightly_smiling_face:

fgaz commented on pkgs/applications/audio/bambootracker/default.nix:
```diff
@@ -0,0 +1,55 @@
+    sha256 = "06ij0k5sfns1zakl6qbxr5j91dj1kfj3s5jc3mpqzxc3zpc3avkv";
+  };
+
+  preConfigure = "cd BambooTracker";
```
Why not `sourceRoot`?

OPNA2608 replied:
Good point, fixed.

fgaz commented on pkgs/applications/audio/bambootracker/default.nix:
```diff
@@ -0,0 +1,55 @@
+    cp -at     $out/share/BambooTracker/ ../demos
+
+    for size in 16x16 256x256; do
+      install res/icon/icon_$size.png $out/share/icons/hicolor/$size/apps/BambooTracker.png
```
`install -m644` on non-executables.

You may also want to use `-D` and skip the `-d` step, so those lines don't need to be kept in sync

OPNA2608 replied:
Gonna change it to `install -Dm644` for the icons & desktop file.

I saw `installManPage` being used in a recent PR, should I use that instead for the manpages?

For the demos, I don't know how to `install` a whole directory with subdirectories. Should I just `chmod 644 -R $out/share/BambooTracker/demos` after `cp`ing them over?

fgaz replied:
> installManPage

I never had to install a manpage yet, but I guess if something like this exists, using it is a goo idea.

> a whole directory
Just `cp -r` is fine afaik. No need to chmod unless the dev set them wrong in the first place I think.

OPNA2608 replied:
Fixed.

fgaz commented on pkgs/applications/audio/bambootracker/default.nix:
```diff
@@ -0,0 +1,54 @@
+  postInstall = ''
+    install -Dm644 ../BambooTracker.desktop $out/share/applications/BambooTracker.desktop
+    installManPage ../BambooTracker*.1
+    cp -r ../demos $out/share/BambooTracker/
```
imo keeping the `demos` subdirectory (`cp -r ../demos $out/share/BambooTracker/demos`) would be a bit nicer than having its contents copied directly in `share/BamboTracker`, but this is fine too

OPNA2608 replied:
Whoops, that was my intention. Fixed.

OPNA2608 commented:
Updated to new v0.4.0 release.

OPNA2608 commented:
Noticed that I missed the i18n & skin files, added those. Does copying all the licensing files to `share/BambooTracker` make sense or is that abit too excessive?

fgaz commented:
I'd like to know that too. In my packages I didn't bother to include them if the makefile/whatever didn't do it by itself, but maybe I should have.

OPNA2608 commented:
Updated to new v0.4.1 release.

We added a workaround for the missing installation of i18n files on our side (it's caused by nixpkgs' qtbase version, QTBUG-77398), so I removed that part of the postInstall phase.

OPNA2608 commented:
Bumped again to v0.4.2.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review-may-2019/3032/175

OPNA2608 commented:
Bumped to v0.4.3.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/195

fgaz commented:
Let's see if this attracts some committer :)

OPNA2608 commented:
~~Thanks for reminding me that this PR is still open.~~

I added a patch to fix the Darwin build, it can be dropped with the next release. Additionally, I ran `nixpkgs-fmt` on the whole file.

danieldk commented on pkgs/applications/audio/bambootracker/default.nix:
```diff
@@ -0,0 +1,62 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A tracker for YM2608 (OPNA) which was used in NEC PC-8801/9801 series computers.";
```
```suggestion
    description = "A tracker for YM2608 (OPNA) which was used in NEC PC-8801/9801 series computers";
```

Description should not end in a dot (following the contribution guidelines).

danieldk commented on pkgs/applications/audio/bambootracker/default.nix:
```diff
@@ -0,0 +1,62 @@
+  meta = with stdenv.lib; {
+    description = "A tracker for YM2608 (OPNA) which was used in NEC PC-8801/9801 series computers.";
+    homepage = "https://github.com/rerrahkr/BambooTracker";
+    license = licenses.gpl2;
```
Should be

```suggestion
    license = licenses.gpl2Only;
```

if this is GPL2 only or `gpl2Later` if it's GPL2 or any later version of the GPL.

danieldk commented on pkgs/applications/audio/bambootracker/default.nix:
```diff
@@ -0,0 +1,62 @@
+    sha256 = "1xkiqira1kpcqkacycy0y7qm1brhf89amliv42byijl4palmykh2";
+  })];
+
+  postPatch = "cd BambooTracker";
+
+  # sourceRoot = "source/BambooTracker";
```
Any reason `sourceRoot` is not used?

OPNA2608 replied:
Makes patch from GitHub commit fail.

```
patching sources
applying patch /nix/store/05w1rpcw9j4y1wik5bqblhc4vnzhmdkv-45346ed99559d44c2e32a5c6138a0835b212e875.patch
can't find file to patch at input line 3
```

danieldk reviewed (changes requested):
Jikes, I am sorry that merging this took so long! I added some small comments so that the derivation follows the guidelines. It would also be nice if you could squash the two commits into one.

Feel free to ping me when you have incorporated these changes, so that we can get this merged!

OPNA2608 committed:
bambootracker: init at 0.4.3

OPNA2608 commented:
@danieldk 

- Rebased so the new license value can be used (this predated that change by months)
- Squashed commits
- Fixed the extra dot

Can't use `sourceRoot` due to the patch.

OPNA2608 commented:
Am I doing something wrong btw? GitHub CI is flooding me with "Run failed" emails.

danieldk commented:
Thanks for making the changes! Not sure where the GitHub CI errors are coming from. But ofborg is happy.

OPNA2608 commented:
Thanks for merging! :+1:
